### ISSUE TITLE
In Browse mode, The form fields such as (checkbox, radio buttons) visible label focused separately and announced separately from the control.

### ISSUE THREAD

**Issue by chikuJ** [nvaccess/nvda#16319]
### Steps to reproduce:
1. Using NVDA+Firefox, Open this URL https://codepen.io/jerrylow/pen/BzPZew
2. Use the down arrow key to focus or navigate through the checkboxes and observe the announcement.

### Actual behavior:
With browse mode navigation, the focus first moves to the checkbox control and is announced as 'Checkbox, <state>' without an accessible label and after pressing the down arrow key again, the focus moves to the visible label and is announced.

### Expected behavior:
As suggested in the https://webaim.org/techniques/forms/controls#checkbox, using the arrow keys, the focus moves only once to the checkbox and is announced along with the label, role, and state.

#### NVDA installed/portable/running from source:
https://www.nvaccess.org/download/

#### NVDA version:
NVDA 2023.3.4

#### Windows version:
Win 11

#### Name and version of other software in use when reproducing the issue:
Mozilla Firefox Version 123.0.1 (64-bit)

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, Tried on NVDA Version 2020

#### If NVDA add-ons are disabled, is your problem still occurring?
NA

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by Brian1Gaff** [comment]
This seems to depend on how the page was designed.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message ----- 
  From: chikuJ 
  To: nvaccess/nvda 
  Cc: Subscribed 
  Sent: Monday, March 18, 2024 6:57 AM
  Subject: [nvaccess/nvda] In Browse mode, The form fields such as (checkbox, radio buttons) visible label focused separately and announced separately from the control. (Issue #16319)



  Each of the questions and sections below start with multiple hash symbols (#). Place your answers and information on the blank line below each question.
  -->

  Steps to reproduce:
    1.. Using NVDA+Firefox, Open this URL https://codepen.io/jerrylow/pen/BzPZew 
    2.. Use the down arrow key to focus or navigate through the checkboxes and observe the announcement. 
    3.. 
  Actual behavior:
  With browse mode navigation, the focus first moves to the checkbox control and is announced as 'Checkbox, ' without an accessible label and after pressing the down arrow key again, the focus moves to the visible label and is announced.

  Expected behavior:
  As suggested in the https://webaim.org/techniques/forms/controls#checkbox, using the arrow keys, the focus moves only once to the checkbox and is announced along with the label, role, and state.

  NVDA installed/portable/running from source:
  https://www.nvaccess.org/download/

  NVDA version:
  NVDA 2023.3.4

  Windows version:
  Win 11

  Name and version of other software in use when reproducing the issue:
  Mozilla Firefox Version 123.0.1 (64-bit)

  Other questions
  Does the issue still occur after restarting your computer?
  Yes

  Have you tried any other versions of NVDA? If so, please report their behaviors.
  Yes, Tried on NVDA Version 2020

  If NVDA add-ons are disabled, is your problem still occurring?
  NA

  Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
  Yes

**Comment by Adriani90** [comment]
Closing as duplicate of #12588.
